File: Python/Programmers/Level 1/나누어 떨어지는 숫자 배열.py
# def sort(arr): # 정렬 구현
#     for i in range(len(arr)):
#         for j in range(i, len(arr)):
#             if arr[i] > arr[j]:
#                 arr[i], arr[j] = arr[j], arr[i]
    
#     return arr

# def solution(arr, divisor):
#     answer = [num for num in arr if num % divisor == 0] # divisor로 나눠지는 숫자만 answer 리스트 생성
#     return [-1] if len(answer) == 0 else sort(answer) # 나눠지는 숫자가 없다면 -1 아니면 answer 정렬

def solution(arr, divisor):
    return sorted([a for a in arr if a % divisor == 0]) or [-1] # or을 사용해 나눠지는 숫자 리스트 또는 [-1]을 

File: JavaScript/Programmers/Level 1/나누어 떨어지는 숫자 배열.js
// arr에서 모든 요소를 divisor로 나누었을 때 나누어 떨어지는 것만 필터링 즉, filter 메소드 이용
// 필터링 된 배열의 길이가 1 이상이면 sort, 0이면 [-1] 반환

function solution(arr, divisor) {
    const answer = arr.filter((el) => !(el % divisor)); // divisor로 나누어 떨어지는 요소만 필터링
    return answer.length ? answer.sort((a, b) => a - b) : [-1]; // answer가 빈배열이 아니면 오름차순 정렬, 빈 배열이면 [-1] return
}
